### PR TITLE
Challenge mod: Deadly Zombie Virus

### LINKED ISSUE #22618: No Cure mod
Feature suggestion:
When infected bites were added, they were carefully balanced based on their inclusion being mandatory and not unduly disrupting the game.  However some people might prefer a more challenging and thematic impact for zomie bites if they opt into it, i.e. the No Cure zombie infection as featured in many zombie movies.  A zombie bite won't be an instant death, but it will be certain, leaving the player to either quit or live out the remainder of their game with no hope of recovery.

Implementation should be fairly simple, add a set of effects parallell to the deep/infected bite effects, sharing most of the same code, with the only difference being that the new effects are not added to the effects of items that cure regular zombie bites.  This would include all treatments chemical and physical, as well as the periodic chance of natural recovery.

### PR DESCRIPTION
**Pull request by dseguin**
#### Summary
None

#### Purpose of change
For those that want to try a more conventional "zombie virus" challenge, bites from zombies have a chance of inflicting a permanent deadly illness. The disease progressively gets worse over the course of 4 days, but the survivor can alleviate their symptoms and extend their life by taking antivirals.

#### Describe the solution
(All content here is limited to the `deadly_bites` mod)

New effects:
- `zombie_virus` - The deadly disease. Grows in intensity over 4 days (without treatment), with the final intensity having a 0.001% chance to kill the player every turn.
- `antivirals` - Prevents the disease from progressing for the duration of the effect.

New item:
- `antivirals` - Medication that adds the `antivirals` effect and reduces the disease's intensity by 1 (if the `antivirals` effect isn't present already).
av_item

New scenario:
- `"Challenge - Deadly Virus Infection"` - Adds the `zombie_virus` effect to the character and sets the current mission to find antivirals.
scenario
mission

##### Mechanics

The zombie's bite attack has an infection chance (usually 10) out of 20, so generally 50%, as opposed to the standard 10% chance to cause a deep bite wound.

infect_msg

The survivor can take an antiviral every day before sleeping to reduce the effect of the disease. They would need to constantly find new antivirals in locations like hospitals and pharmacies to survive.

av_msg

These are pretty much all just JSON additions, plus a little bit of code necessary to get the mechanics working.

#### Describe alternatives you've considered
Dropping the idea, I don't think most folks would be interested anyway.

#### Testing
TODO:
- Tweak infection chance and spawn rate for antivirals :heavy_check_mark:
  - Infection chance is high, ~50% after a bite (as intended, just avoid getting grabbed)
  - Antivirals spawn pretty reliably in pharmacies and hospitals, with occasional spawns in doctor's offices and vet clinics
- A little more playtesting :heavy_check_mark:

#### Additional context
(closes #22618)

### CODE REVIEW

File: src/character.cpp [hunk 1]
```diff
@@ -3248,7 +3248,7 @@ void Character::normalize()
 void Character::die( Creature *nkiller )
 {
     g->set_critter_died();
-    is_dead = true;
+    set_all_parts_hp_cur( 0 );
```
Comment (anothersimulacrum) [inline src/character.cpp:3251]: Why this change? If you are doing this, you should probably just remove the variable entirely (I know of no other uses).

Comment (dseguin) [reply src/character.cpp:3251]: Cheers, I'll just remove it. It turns out that `Character`s are only considered "dead" when a vital bodypart reaches 0 hp, and that `is_dead` was doing nothing as far as I could tell.

**Comment by NetSysFire** [comment]
Cool! But if it is not curable, why does this exist? Is this just something that will never get triggered but is still mandatory?

https://github.com/CleverRaven/Cataclysm-DDA/blob/279aba091d6476437b619d6853926c63064c1c7c/data/mods/deadly_bites/effects.json#L26

**Comment by NetSysFire** [comment]
Ah, good. :)

Also, please consider adding a PR summary instead of using None when this is ready for merging because this is a feature which should definitely be in the changelog.

**Comment by Night-Pryanik** [comment]
> I don't think most folks would be interested anyway

Can't say for others, but I'm definitely interested.

**Comment by schoolcf** [comment]
would be nice if antivirals also cured other virii

**Comment by dseguin** [comment]
> [...] and this code fits that purpose(-no drug to stave off infection), I think the code should also be used for that.

That's a good point, but I think it might be better to add the changes to DDotD as a separate PR

> would be nice if antivirals also cured other virii

No problem. The antivirals now block the flu and the common cold (the two other viral diseases that I know of). I guess that makes these "broad-spectrum antivirals" (if such a thing exists)

**Comment by kevingranade** [comment]
> I guess that makes these "broad-spectrum antivirals" (if such a thing exists)

It does, and they don't. Please limit the scifi medicine to working on scifi diseases.

File: data/mods/deadly_bites/effects.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+[
+  {
+    "type": "effect_type",
+    "id": "antivirals",
+    "rating": "good",
+    "blocks_effects": [ "zombie_virus", "flu", "common_cold" ],
```
Comment (dseguin) [inline data/mods/deadly_bites/effects.json:6]:
```suggestion
    "blocks_effects": [ "zombie_virus" ],
```
